docs/src/test/kotlin/DocSnippetTests.kt

```diff
@@ -150,6 +150,7 @@ class DocSnippetTestsEngine : HierarchicalTestEngine<DocSnippetTestsEngine.Execu
     }
   }
 
+  @Suppress("DEPRECATION")
   private fun parseAsciidoc(docDescriptor: Descriptor.Path, selectors: List<UniqueIdSelector>) {
```

Comment:
Certain methods of `AbstractDeque` produce warnings, but they are just used here in tests, and the warnings involved describe missed optimizations. So I have suppressed them.